### PR TITLE
8307299: Move more DnD tests to open

### PR DESCRIPTION
**Pull request by alisenchung**
Added DnD tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307299](https://bugs.openjdk.org/browse/JDK-8307299): Move more DnD tests to open

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [ac214c9f](https://git.openjdk.org/jdk/pull/13759/files/ac214c9fae51c2793093429b8215988d967d2ebf)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [0f5df266](https://git.openjdk.org/jdk/pull/13759/files/0f5df266d7b6fea520db32ab5312c52432bd6ef3)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13759/head:pull/13759` \
`$ git checkout pull/13759`

Update a local copy of the PR: \
`$ git checkout pull/13759` \
`$ git pull https://git.openjdk.org/jdk.git pull/13759/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13759`

View PR using the GUI difftool: \
`$ git pr show -t 13759`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13759.diff">https://git.openjdk.org/jdk/pull/13759.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13759#issuecomment-1531940640)

### CODE REVIEW

File: test/jdk/java/awt/dnd/DropTargetingTest.java [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
```
Comment (prrace) [inline test/jdk/java/awt/dnd/DropTargetingTest.java:25]: This isn't important and I noticed the same on the other PR I just looked at, but generally
we'd list javax.* APIs after the java.* APIs 

But it isn't 100% alphabetical as I'd put all the "java.base" APIs first ... not relevant here tho'

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        robot.waitForIdle();
+        reset();
+        robot.mouseMove(p.x, p.y);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
+        synchronized (SYNC_LOCK) {
+            robot.mouseRelease(InputEvent.BUTTON1_MASK);
+            SYNC_LOCK.wait(MOUSE_RELEASE_TIMEOUT);
+        }
+
+        Component c = clickedComponent;
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java:275]: this field is used on EDT and the main thread, some synchronisation is needed.

File: test/jdk/java/awt/dnd/LinkAcceptanceTest.java [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        LinkAcceptanceTest test = new LinkAcceptanceTest();
+        if (args.length > 0) {
+            test.run(args);
+        } else {
+            EventQueue.invokeAndWait(test::init);
+            try {
+                test.start();
+            } finally {
+                EventQueue.invokeAndWait(() -> {
+                    if(test.frame != null) {
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/LinkAcceptanceTest.java:94]: missed space after "if"

File: test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java (hunk 2 above)
Comment (alisenchung) [reply test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java:275]: Should I put that field on the EDT as well?

File: test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        }
+
+        if (dragSourceListener.getDropSuccess() != success) {
+            throw new RuntimeException("getDropSuccess() returned wrong value:"
+                                       + dragSourceListener.getDropSuccess());
+        }
+    }
+
+    public void reset() {
+        clickedComponent = null;
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java:253]: seems here you access/reset that field on the main thread as well

File: test/jdk/java/awt/dnd/DroppingVMHangTest.java [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+        DroppingVMHangTest test = new DroppingVMHangTest();
+        if (args.length > 0) {
+            test.run(args);
+        } else {
+            EventQueue.invokeAndWait(test::init);
+            try {
+                test.start();
+            } finally {
+                EventQueue.invokeAndWait(() -> {
+                    if(test.frame != null) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/DroppingVMHangTest.java:98]:
```suggestion
                    if (test.frame != null) {
```

File: test/jdk/java/awt/dnd/InterJVMLinkTest.java [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+        InterJVMLinkTest test = new InterJVMLinkTest();
+        if (args.length > 0) {
+            test.run(args);
+        } else {
+            EventQueue.invokeAndWait(test::init);
+            try {
+                test.start();
+            } finally {
+                EventQueue.invokeAndWait(() -> {
+                    if(test.frame != null) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/InterJVMLinkTest.java:85]:
```suggestion
                    if (test.frame != null) {
```

File: test/jdk/java/awt/dnd/DropTargetingTest.java [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+class DropTargetPanel extends Panel implements DropTargetListener {
+
+    final Dimension preferredDimension = new Dimension(200, 100);
+    boolean testPassed = true;
+
+    public DropTargetPanel() {
+        setDropTarget(new DropTarget(this, this));
+    }
+
+    public boolean getStatus() {
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/DropTargetingTest.java:256]: This method(and probably some others) are not used?

**Review by mrserb — APPROVED** [review]
Looks fine, please confirm march5 is green.

**Pull request closed without merging**